driver/infoblox_ipam.go

```diff
@@ -76,10 +76,28 @@ func (ibDrv *InfobloxDriver) RequestAddress(r *ipamApi.RequestAddressRequest) (*
 	if rType, ok := r.Options["RequestAddressType"]; macAddr == "" || (ok && rType == netlabel.Gateway) {
 		logrus.Debugln("Request for Gateway IP")
 
+		l := &ibclient.NetworkViewLock{Name: network.NetviewName, ObjMgr: ibDrv.objMgr, LockEA: common.EA_DOCKER_PLUGIN_LOCK,
+			LockTimeoutEA: common.EA_DOCKER_PLUGIN_LOCK_TIME}
+		err := l.Lock()
+
+		if err != nil {
+			return  &ipamApi.RequestAddressResponse{}, fmt.Errorf("Failed to Allocate Gateway IP :%s", err)
+		}
+		defer l.UnLock(false)
+
+		if r.Address != "" {
+			ipAddr, err := ibDrv.objMgr.GetFixedAddress(network.NetviewName, network.Cidr, r.Address, "")
+			if err == nil && ipAddr != nil {
```

Comment:
Right, I know you cannot return the MAC as part of the RequestAddress response.

OK, it starts coming back now... so I believe the main purpose of the  --gateway option is just so that it can be injected into the container so it can configure the gateway ip accordingly.

Thank you for your response - I'm good!